exist-core/src/main/java/org/exist/xslt/XSLTErrorsListener.java

```diff
@@ -31,64 +32,70 @@
  */
 public abstract class XSLTErrorsListener<E extends Exception> implements ErrorListener {
 
-  private final static Logger LOG = LogManager.getLogger(XSLTErrorsListener.class);
+  private static final Logger LOG = LogManager.getLogger(XSLTErrorsListener.class);
 
-  private final static int NO_ERROR = 0;
-  private final static int WARNING = 1;
-  private final static int ERROR = 2;
-  private final static int FATAL = 3;
+  private enum ErrorType {
+    WARNING, ERROR, FATAL;
+  }
 
-  boolean stopOnError;
-  boolean stopOnWarn;
+  private final boolean stopOnError;
+  private final boolean stopOnWarn;
 
-  private int errorCode = NO_ERROR;
-  private Exception exception;
+  @Nullable private ErrorType errorType;
+  @Nullable private TransformerException exception;
 
-  public XSLTErrorsListener(boolean stopOnError, boolean stopOnWarn) {
+  public XSLTErrorsListener(final boolean stopOnError, final boolean stopOnWarn) {
     this.stopOnError = stopOnError;
     this.stopOnWarn = stopOnWarn;
   }
 
-  protected abstract void raiseError(String error, Exception ex) throws E;
+  protected abstract void raiseError(String error, TransformerException ex) throws E;
 
   public void checkForErrors() throws E {
-    switch (errorCode) {
+    if (errorType == null) {
+      return;
+    }
+
+    switch (errorType) {
       case WARNING:
         if (stopOnWarn) {
-          raiseError("XSL transform reported warning: " + exception.getMessage(), exception);
+          raiseError("XSL transform reported warning: " + exception.getMessageAndLocation(), exception);
         }
         break;
       case ERROR:
         if (stopOnError) {
-          raiseError("XSL transform reported error: " + exception.getMessage(), exception);
+          raiseError("XSL transform reported error: " + exception.getMessageAndLocation(), exception);
         }
         break;
       case FATAL:
-        raiseError("XSL transform reported error: " + exception.getMessage(), exception);
+        raiseError("XSL transform reported error: " + exception.getMessageAndLocation(), exception);
     }
```

Comment:
Could you add the missing break here?

Reply:
@reinhapa I would like to keep this PR the same as the other two that were merged if possible. The missing break there will not actually cause any harm.